liblangutil/CharStream.cpp

```diff
@@ -118,3 +118,29 @@ tuple<int, int> CharStream::translatePositionToLineColumn(int _position) const
 	}
 	return tuple<int, int>(lineNumber, searchPosition - lineStart);
 }
+
+std::optional<int> CharStream::translateLineColumnToPosition(int _line, int _column) const
+{
+	int line = 0;
+	int column = 0;
+	size_t offset = 0;
+
+	while (offset < m_source.size())
```

Comment:
Is this function a big performance bottleneck? Would it be faster when done using `for (char c: m_source) { ...} `?

Reply:
I tried it I don't see any benefit in using a range based for loop here, but I rewrote it nevertheless by first finding the line number by looping over `offset = m_source.find(LF, offset + 1)`, and then just adding `_column` to it. I think if at all, then this one is faster (ideally with `find(char, offset)` providing a vectorized impl :-)

Reply:
Did you profile it?